genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/TitusAgentLauncher.java

```diff
@@ -0,0 +1,226 @@
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Health health() {
+        return Health.up()
+            .withDetails(this.healthIndicatorCache.asMap())
+            .build();
+    }
+
+    private TitusBatchJobRequest createJobRequest(final ResolvedJob resolvedJob) {
+
+        final String jobId = resolvedJob.getJobSpecification().getJob().getId();
+
+        // Map placeholders in entry point template to their values
+        final Map<String, String> placeholdersMap = ImmutableMap.of(
+            TitusAgentLauncherProperties.JOB_ID_PLACEHOLDER,
+            jobId,
+            TitusAgentLauncherProperties.SERVER_HOST_PLACEHOLDER,
+            this.titusAgentLauncherProperties.getGenieServerHost(),
+            TitusAgentLauncherProperties.SERVER_PORT_PLACEHOLDER,
+            String.valueOf(this.titusAgentLauncherProperties.getGenieServerPort())
+        );
+
+        // Substitute all placeholders with their values
+        final List<String> entryPoint =
+            this.titusAgentLauncherProperties.getEntryPointTemplate()
+                .stream()
+                .map(s -> placeholdersMap.getOrDefault(s, s))
+                .collect(Collectors.toList());
+
+        return new TitusBatchJobRequest(
+            new TitusBatchJobRequest.Owner(this.titusAgentLauncherProperties.getOwnerEmail()),
+            this.titusAgentLauncherProperties.getApplicationName(),
+            this.titusAgentLauncherProperties.getCapacityGroup(),
+            ImmutableMap.of(
+                GENIE_USER_ATTR, resolvedJob.getJobMetadata().getUser(),
+                GENIE_SOURCE_HOST_ATTR, this.genieHostInfo.getHostname(),
+                GENIE_ENDPOINT_ATTR, this.titusAgentLauncherProperties.getGenieServerHost(),
+                GENIE_JOB_ID_ATTR, jobId
+            ),
+            new TitusBatchJobRequest.Container(
+                new TitusBatchJobRequest.Resources(
+                    resolvedJob.getJobEnvironment().getCpu(),
+                    0,
+                    resolvedJob.getJobEnvironment().getMemory(),
+                    this.titusAgentLauncherProperties.getDiskSize().toMegabytes(),
+                    this.titusAgentLauncherProperties.getNetworkBandwidth().toMegabytes() * 8 //MB to Mb
+                ),
+                new TitusBatchJobRequest.SecurityProfile(
+                    this.titusAgentLauncherProperties.getSecurityAttributes(),
+                    this.titusAgentLauncherProperties.getSecurityGroups(),
+                    this.titusAgentLauncherProperties.getIAmRole()
+                ),
+                new TitusBatchJobRequest.Image(
+                    this.titusAgentLauncherProperties.getImageName(),
+                    this.titusAgentLauncherProperties.getImageTag()
+                ),
+                entryPoint,
+                this.titusAgentLauncherProperties.getAdditionalEnvironment()
+            ),
+            new TitusBatchJobRequest.Batch(
+                1,
```

Comment:
this `1` is a magic number with no real context to it, perhaps a constant would be helpful to know what it means?